packages/ember-page-layout/src/components/page/sidebar.ts

```diff
@@ -0,0 +1,66 @@
+import { getOwner } from '@ember/application';
+import { action } from '@ember/object';
+import { guidFor } from '@ember/object/internals';
+import Router from '@ember/routing/router-service';
+import { scheduleOnce } from '@ember/runloop';
+import { inject as service } from '@ember/service';
+import Component from '@glimmer/component';
+import { tracked } from '@glimmer/tracking';
+
+import Layout from '../../services/layout';
+
+export interface PageSidebarArgs {
+    id?: string;
+    empty?: boolean;
+    document?: Document;
+}
+
+export interface PageSidebarSignature {
+    Args: PageSidebarArgs;
+    Blocks: { default: []; initial: [] };
+}
+
+export default class PageSidebar extends Component<PageSidebarSignature> {
```

Comment:
not a dealbreaker, but seems like the `header`, `footer`, and `sidebar` component implementations are almost similar, thoughts on just making them extend a `BasePageComponent` base component, with a couple properties to override? e.g.
```ts
class PageSidebar extends BasePageComponent<PageSidebarSignature> {
  defaultId = 'application-sidebar';
  layoutTreeProperty = 'sidebarTree';
}
```

Reply:
Short answer - yes. I want to actually make it globally available so that you can use it with any name. But not for this initial commit